Firestore/tests/System/AggregateQueryTest.php

```diff
@@ -103,15 +124,39 @@ private function insertDocs(array $docs)
         return $docsRefs;
     }
 
+    private function compareResult($expected, $actual)
+    {
+        if (is_nan($expected)) {
```

Comment:
In php8.2, this might give deprecation notices / warnings (not exactly sure), for passing null into `is_nan`

Reply:
Probably this needs to be fixed here as well (by reversing the `is_nan` and `is_null` checks):
https://github.com/googleapis/google-cloud-php/blob/8c7e171ba8d85b90a2dd7d5a7b9a9b7b38fc095f/Firestore/src/Query.php#L1040-L1047

Reply:
Raised https://github.com/googleapis/google-cloud-php/pull/6771 to fix `Query.php`, this test also needs a fix accordingly.

Reply:
Fixed by adding a null check beforewards:

```
if (!is_null($expected) && is_nan($expected)) {}
else if(is_float($expected)) {}
else {}
```

Nothing changes due to this as even if the `$result` is `null`, the code flow would enter the last else.